controllers/rack.go

```diff
@@ -587,14 +638,15 @@ func (r *SingleClusterReconciler) upgradeRack(
 		podsToUpgrade = append(podsToUpgrade, pod)
 	}
 
-	// Failed pods should always come before active pods.
-	// We make info calls before restarting active pods,
-	// if active pods are checked before failed pods then
-	// info call will fail in failed pods and operator will not go ahead
-	rearrangedPods := getRearrangedFailedAndActivePods(podsToUpgrade)
+	var podsBatchList [][]*corev1.Pod
 
-	// Create batch of pods
-	podsBatchList := r.getPodsBatchToRestart(rearrangedPods, len(podList))
+	if len(failedPods) != 0 {
```

Comment:
nit: would be better to add a comment why we are doing this. Also we can add a log saying skipping batchSize for failed pod or something related